### PR TITLE
Locate mainFile using mainDirname

### PR DESCRIPTION
**Pull request by HeilTec**
Greetings and thank you for making this package!

I found these changes were needed on my most simplistic setup. 

The app.getAppPath() function returns the distribution package in:
   `/usr/local/lib/node_modules/electron/dist/resources/default_app.asar`

When just using the command
  `electron .`
to start the app, then saving a new version of the main application file ('main.js') does not update this file.
I hope that just watching the `main.js` source file should be sufficient in more advances settings.

I left in the commented console.log's as they might be useful in later debugging.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -36,30 +36,43 @@ module.exports = (glob, options) => {
   let eXecutable = options.electron;
   if(eXecutable && fs.existsSync(eXecutable)) {
     let appPath = app.getAppPath();
-    // eslint-disable-line
+
+		// console.log('electron-reload: appPath:', appPath);
```
Comment (yan-foto) [inline main.js]: This doesn't look good!

File: main.js [hunk 2]
```diff
@@ -4,11 +4,11 @@ const fs = require('fs');
 const {spawn} = require('child_process');
 const path = require('path');
 
-module.exports = (glob, options) => {
+module.exports = (mainDirname, options) => {
   options = options || {};
   let browserWindows = [];
   let opts = Object.assign({ignored: /node_modules|[\/\\]\./}, options);
-  let watcher = chokidar.watch(glob, opts);
+  let watcher = chokidar.watch(mainDirname, opts);
```
Comment (yan-foto) [inline main.js]: Is there any reason to change `glob` to `mainDirname`?

File: main.js [hunk 3]
```diff
@@ -36,30 +36,43 @@ module.exports = (glob, options) => {
...
-      // firing an `app.quit()` may not actually quit the app. In these cases
-      // you can use `app.exit()` to gracefully close the app.
-      if(opts.hardResetMethod === 'exit'){
-          app.exit();
-      } else {
-          app.quit();
-      }
-    });
-  } else {
+		if (fs.existsSync(mainFile)) {
```
Comment (yan-foto) [inline main.js]: From this block on, the whole formatting has changed and is not possible to see the actual changes.

Comment (yan-foto) [inline main.js]: Why do you check for a `mainFile`? If there is no main file, there is no electron package 🤔

**Comment by yan-foto** [comment]
This issue has already been discussed in #17 . I hadn't been able to reproduce this problem locally and have closed that issue. If you have information on how to reproduce, please provide them before I can consider this pull request.

**Comment by yan-foto** [comment]
**Update**: I just followed electron's [official quickstart](https://github.com/electron/electron/blob/master/docs/tutorial/quick-start.md) to bootstrap a project, fetched the latest version of `electron-reload` from npm and tested hard-reset locally and everything worked as expected.

So I'm interested to know how your "most simplistic" setup looks like.

**Comment by HeilTec** [comment]
I can see now that my "solution" is not the right one, but since you asked to see what my simplistic setup looks like I made a [gist](https://gist.github.com/HeilTec/defde2387e7cc03537ac55855193728a.js) of it including a reformatted patched version if reload.
I use VS Code and start my test app using the console command: npm start
My OS is Ubuntu 16

**Comment by yan-foto** [comment]
At least we have it all documented now 👍 . Thanks for contributing.

**Comment by HeilTec** [comment]
I really like the basic concepts of this project.
I am still baffled by inconsistent result from calling `app.getAppPath()`

At first I thought it was a VS Code issue since I could see that starting from a terminal seemed to work

vs code restart
VS CODE

terminalrestart gimp
Terminal

Sadly I must confess to inexperience on setting up shell environment. Further experiments reverted to failing results. If you have any ideas for me to get wiser I would be very appreciative.

**Comment by yan-foto** [comment]
Keep up the good work, don't be too hasty when it comes to pull requests, read the source thoroughly, and provide as much as (relevant) information when opening an issue. ✌️

**Pull request closed without merging**